code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,403 @@
+			return
+
+		visible_message("<span class='notice'>The console beeps happily as it disgorges \the [I].</span>")
+		I.forceMove(drop_location())
+		frozen_items -= I
+
+	else if(href_list["allitems"])
+		if(!allowed(user))
+			to_chat(user, "<span class='warning'>Access Denied.</span>")
+			return
+		if(!allow_items) return
+
+		if(frozen_items.len == 0)
+			to_chat(user, "<span class='notice'>There is nothing to recover from storage.</span>")
+			return
+
+		visible_message("<span class='notice'>The console beeps happily as it disgorges the desired objects.</span>")
+
+		for(var/obj/item/I in frozen_items)
+			I.forceMove(drop_location())
+		frozen_items.Cut()
+
+	updateUsrDialog()
+	return
+
+//Cryopods themselves.
+/obj/machinery/cryopod
+	name = "cryogenic freezer"
+	desc = "Suited for Cyborgs and Humanoids, the pod is a safe place for personnel affected by the Space Sleep Disorder to get some rest."
+	icon = 'icons/obj/machines/sleeper.dmi'
+	icon_state = "cryopod-open"
+	density = TRUE
+	anchored = TRUE
+	state_open = TRUE
+	var/ready = FALSE
+
+	var/on_store_message = "has entered long-term storage."
+	var/on_store_name = "Cryogenic Oversight"
+
+	// 5 minutes-ish safe period before being despawned.
+	var/time_till_despawn = 5 MINUTES // This is reduced to 30 seconds if a player manually enters cryo
+
+	var/obj/machinery/computer/cryopod/control_computer
+	var/cooldown = FALSE
+
+/obj/machinery/cryopod/Initialize()
+	..()
+	GLOB.cryopods += src
+	return INITIALIZE_HINT_LATELOAD //Gotta populate the cryopod computer GLOB first
+
+/obj/machinery/cryopod/Destroy()
+	GLOB.cryopods -= src
+	..()
+
+/obj/machinery/cryopod/LateInitialize()
+	update_icon()
+	find_control_computer()
+
+/obj/machinery/cryopod/proc/find_control_computer(urgent = 0)
+	for(var/M in GLOB.cryopod_computers)
+		var/obj/machinery/computer/cryopod/C = M
+		if(get_area(C) == get_area(src))
+			control_computer = C
+			break
+
+	// Don't send messages unless we *need* the computer, and less than five minutes have passed since last time we messaged
+	if(!control_computer && urgent && !cooldown)
+		cooldown = TRUE
+		log_admin("Cryopod in [get_area(src)] could not find control computer!")
+		message_admins("Cryopod in [get_area(src)] could not find control computer!")
+		addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 5 MINUTES)
+
+	return control_computer != null
+
+/obj/machinery/cryopod/close_machine(mob/user)
+	if(!control_computer)
+		find_control_computer(TRUE)
+	if((isnull(user) || istype(user)) && state_open && !panel_open)
+		..(user)
+		var/mob/living/mob_occupant = occupant
+		if(mob_occupant && mob_occupant.stat != DEAD)
+			to_chat(occupant, "<span class='boldnotice'>You feel cool air surround you. You go numb as your senses turn inward.</span>")
+		var/offer = alert(mob_occupant, "Do you want to offer yourself to ghosts?", "Ghost Offer", "Yes", "No")
+		if(offer == "Yes" && offer_control(occupant))
```

Comment:
ahhhhhhhhhhhhhhh check if the user is still in the machine

Reply:
jamie went zzzzz

Reply:
> ahhhhhhhhhhhhhhh check if the user is still in the machine

I DID THAT, YOU ASKED ME TO REMOVE IT! REEEEEEEEEEEEEEEEEEEEEEEEEEEEE

Reply:
i Asked you to remove the message, not the check